### DISCUSSION TITLE
Link the port end

### DISCUSSION THREAD

**Discussion by ITfanheng** [clientIO/joint#1706] [Q&A] [answered]
Hi!

Can i connect two port end?

Here are the electronic components.

<img width="349" alt="截屏2022-06-20 下午5 32 18" src="https://user-images.githubusercontent.com/75559678/174572658-7bc0bfe4-5052-4be0-a7b3-6cf1b0dd465c.png">

Thanks in advance!

**Comment by kumilingus** [comment, marked as answer]
Yes. https://jsfiddle.net/kumilingus/0m2zfthw/

**Reply by ITfanheng** [reply]
Thanks for your reply!

This demo meets most of my needs！！

But if i can link the port end, the app will looks more professional.(Linked as red line)

<img width="347" alt="截屏2022-06-21 上午9 09 46" src="https://user-images.githubusercontent.com/75559678/174696676-96b51084-7ffd-42a8-8aa7-187df61f8b34.png">

**Reply by kumilingus** [reply]
You can use an [anchor](https://resources.jointjs.com/docs/jointjs/v3.5/joint.html#anchors.intro).

https://jsfiddle.net/kumilingus/0m2zfthw/15/

**Reply by ITfanheng** [reply]
Thank you.

This demo is very nice!!

**Comment by ITfanheng** [comment]
Hi,

One more question.

How to link like that manually?

I have try to set the anchor in `validateConnection`

```typescript
validateConnection: (cellViewS, magnetS, cellViewT, magnetT, end, linkView) => {
  if (end === 'target') {
      if (!magnetT) {
          return false;
      }
      if (cellViewS === cellViewT) {
          return false;
      }
  } else {
      if (!magnetS) {
          return false;
      }
  }
  linkView.model.prop('source/anchor', { name: 'left' });
  linkView.model.prop('target/anchor', { name: 'right' });
  return true;
}
```

but only the `source` have succeeded.

<img width="1116" alt="截屏2022-07-20 上午11 32 01" src="https://user-images.githubusercontent.com/75559678/179890672-c8fa1ec5-8556-420c-a365-ce64cafc15da.png">


And when i move the `source` to other port. It goes wrong, And the widget was stuck too.

<img width="1166" alt="截屏2022-07-20 上午11 34 28" src="https://user-images.githubusercontent.com/75559678/179890911-c06ae784-63dd-4ef6-9e68-7362dd172df2.png">

Any help will appreciate!

**Reply by ITfanheng** [reply]
The device maybe have 4 directions port. So link anchor need flexible.

**Reply by kumilingus** [reply]
Don't add side effects to the `validateConnection` callback. It's better to use `connectionStrategy` to modify the source/target on connect (https://resources.jointjs.com/docs/jointjs/v3.5/joint.html#connectionStrategies.custom).

**Reply by ITfanheng** [reply]
Thank you very much!
